src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiAddProjectPropertiesStep.cs

```diff
@@ -68,10 +69,12 @@ protected override async Task<UpgradeStepApplyResult> ApplyImplAsync(IUpgradeCon
                     }
             }
 
-            // Use MAUI tag
-            file.SetPropertyValue("UseMaui", "true");
-            await file.SaveAsync(token).ConfigureAwait(false);
+            if (projectproperties.GetProjectPropertyValue("UseMaui").FirstOrDefault() == null)
```

Comment:
prefer `is null`
Also is there a chance that `GetProjectPropertyValue` comes back null altogether?
```suggestion
            if (projectproperties.GetProjectPropertyValue("UseMaui")?.FirstOrDefault() is null)
```

Reply:
Hmm I think if Property itself is null or value of property is null, both would require the property to be added so it should work out fine. I have updated the syntax to your suggested change :)